Add Picture In Picture /updates article

beaufortfrancois opened the pull request:
WIP @mounirlamouri

Live preview at https://pr-5014-dot-web-central.appspot.com/web/updates/2017/09/picture-in-picture

mounirlamouri commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,93 @@
+- Notify the website when it enters and leave Picture in Picture mode.
+- Allow the website to trigger Picture in Picture via a user gesture on a video element.
+- Allow the website to exit Picture in Picture.
+- Allow the website to check if Picture in Picture can be triggered.
```
On the four points above, can you make it clear that it's the video going PIP? you say "the website" and "it" which can put the wrong expectations.

beaufortfrancois replied:
Done.

johnpallett commented:
Would change explanation and comment on "How to prevent" section, make it more explanatory and less edgy. Was:

_There is one way you can prevent video to go automatically PiP in Chrome while the Web API is not implemented yet._

```
// Nasty hack ahead!
window.addEventListener('resiz

```

Change to:

_Today, you can prevent video from automatic PiP in Chrome by responding to a resize event, and detecting when the video size has changed significantly. This is a not recommended as a permanent solution but provides a temporary option until the Web API is implemented._

```
// See whether resize is small enough to be PiP. It's a hack, but it'll work for now.
window.addEventListener('resiz

```

jondahlke commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,108 @@
+  </figcaption>
+</figure>
+
+It is but... what about desktop? What if the website want to control that
```
It is, [Add comma] but... what about desktop? What if the website wants [add s] to control that experience?

beaufortfrancois replied:
Done.

jondahlke commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,108 @@
+
+The good news is that a [Picture In Picture Web API] specification is being
+drafted as we speak. This spec aims to allow websites to initiate and control
+this behavior by exposing the following sets of properties to the API:
```
Should it be "set of properties" instead of "sets of properties"?

beaufortfrancois replied:
Done.

jondahlke commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,108 @@
+drafted as we speak. This spec aims to allow websites to initiate and control
+this behavior by exposing the following sets of properties to the API:
+
+- Notify the website when video enters and leaves Picture in Picture mode.
```
Notify the website when a [add a] video enters and leaves Picture in Picture mode.

beaufortfrancois replied:
Done.

jondahlke commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,108 @@
+{% include "web/_shared/contributors/beaufortfrancois.html" %}
+
+Since April 2017, Chrome for [Android O supports Picture In Picture]. It allows
+users to play a `<video>` element in a pinned window that isn't blocked by other
```
Is "pinned" window the right word? Perhaps, we say "overlay", "secondary", "small", or "miniature".

beaufortfrancois replied:
Done.

jondahlke commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,108 @@
+
+Here's how it works: Open Chrome, go to a website that contains a video and
+play it fullscreen. From there, press the Home button to go to your Android
+Home Screen and the playing video will automatically transition in Picture In
```
automatically transition to [instead of in] Picture in Picture

beaufortfrancois replied:
Done.

jondahlke commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,108 @@
+So what do you think? Please submit your feedback and raise issues in the
+[Picture In Picture WICG repository]. We're eager to hear your thoughts!
+
+## Prevent Auto PiP
```
+1 to John's comments about the opening to this section. I would add that "Auto PiP" may not be the right branding for that behavior. How about "Preventing Android's default PIP behavior?"

beaufortfrancois replied:
Done.

beaufortfrancois commented:
@johnpallett @jondahlke I've addressed all your feedback. Thank you!

@jpmedley Do you mind having a look and merge if that looks good to you?

beaufortfrancois commented:
@johnpallett @jondahlke I've addressed all your feedback. Thank you!

@jpmedley Do you mind having a look and merge if that looks good to you?

jpmedley commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,109 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Let's discuss about a Web API that would allow websites to create a floating video window over the desktop.
```
Remove 'about'.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,109 @@
+{# wf_published_on: 2017-09-12 #}
+{# wf_tags: news,media #}
+{# wf_featured_image: /web/updates/images/generic/picture-in-picture.png #}
+{# wf_featured_snippet: Let's discuss about a Web API that would allow websites to create a floating video window over the desktop #}
```
Remove 'about'.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,109 @@
+users to play a `<video>` element in a small overlay window that isn't blocked
+by other windows, so that they can watch while doing other things.
+
+Here's how it works: Open Chrome, go to a website that contains a video and
```
'Open' should be lowercase.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,109 @@
+      });
+    </script>
+
+Warning: Code above is not implemented by browsers yet.
```
Change 'Code' to 'The code'.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,109 @@
+
+## Preventing Android's default PIP behavior
+
+Today, you can prevent video from Android's defaut PiP behavior in Chrome by
```
Change 'from' to 'from using'.

Change 'defaut' to 'default'.

Did I miss something? Why do I need to prevent Android's default PiP behavior if I don't have it yet?

beaufortfrancois replied:
Done.
Done.

Android's default PiP behaviour will still be on top of the Web API.
Sadly, as the API doesn't exist yet, there is no proper way to say "no". See https://github.com/WICG/picture-in-picture/issues/25

jpmedley commented on src/content/en/updates/2017/09/picture-in-picture.md:
```diff
@@ -0,0 +1,109 @@
+
+Today, you can prevent video from Android's defaut PiP behavior in Chrome by
+responding to a resize event, and detecting when the window size has changed
+significantly. This is a not recommended as a permanent solution but provides a
```
Define significantly.

Change 'is a not' to 'is not'.

beaufortfrancois replied:
I've pointed to the code below.

Done.

beaufortfrancois commented:
@jpmedley I've addressed your feedback at https://github.com/google/WebFundamentals/pull/5014/commits/0a5c064619503d6fdae94237630ccc8356f2653c. Please merge and push live if that LGTY.
